cvss/scorev3.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// scoring is implemented based on the specification at:
+// https://www.first.org/cvss/specification-document
+
+package cvss
+
+import (
+	"math"
+)
+
+// Severity represents scores severity
+type Severity int
+
+const (
+	SeverityNone Severity = iota
+	SeverityLow
+	SeverityMedium
+	SeverityHigh
+	SeverityCritical
+)
+
+func (s Severity) String() string {
+	switch s {
+	case SeverityNone:
+		return "None"
+	case SeverityLow:
+		return "Low"
+	case SeverityMedium:
+		return "Medium"
+	case SeverityHigh:
+		return "High"
+	case SeverityCritical:
+		return "Critical"
+	default:
+		panic("undefined severity")
+	}
+}
+
+// SeverityFromScore will return the severity assigned to given score
+func SeverityFromScore(score float64) Severity {
+	if score <= 0 {
+		return SeverityNone
+	}
+	if 0 < score && score < 4 {
+		return SeverityLow
+	}
+	if 4 <= score && score < 7 {
+		return SeverityMedium
+	}
+	if 7 <= score && score < 9 {
+		return SeverityHigh
+	}
+	return SeverityCritical
+}
+
+const (
```

Comment:
You use this for both base and modified scores. Specification does not say anywhere these are the same coefficients. I presume, your decision to make them constants was based on the idea that you might want to change them at some point. But it is not guaranteed that you will have to change both and not just one. My point is, just hardcode them, these are the kind of "magic numbers" that actually are "magic numbers" by design. There's no point in giving them names.

Reply:
https://www.first.org/cvss/calculator/cvsscalc30.js
I got that from here. These two are the only constants in their implementation so I thought there's some background here. I personally have no strong opinions about it, so I'll do as you said.

Reply:
I don't have a strong opinion either, I just don't see the point :) I'm fine with both ways.